packages/firestore/src/remote/rest_connection.ts

```diff
@@ -98,7 +99,16 @@ export abstract class RestConnection implements Connection {
     };
     this.modifyHeadersForRequest(headers, authToken, appCheckToken);
 
-    return this.performRPCRequest<Req, Resp>(rpcName, url, headers, req).then(
+    const { host } = new URL(url);
+    const forwardCredentials =
+      isCloudWorkstation(host) && this.databaseInfo.isUsingEmulator;
```

Comment:
I'm wondering if you even need `&& this.databaseInfo.isUsingEmulator`?

Would it be acceptable to only check for `isCloudWorkstation(host)`?

I'm concerned that a user could connect to the emulator without calling `connectFirestoreEmulator(...)`, simply by using FirestoreSettings. `initializeFirestore(app, { host: cloudWorkstationUrl, ssl: true, port: 443 })`.

And it may be the agent that attempts to connect that way

Reply:
Removing `isUsingEmulator`